packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -166,7 +184,16 @@ export class IncomingRequestsController extends ConsumptionBaseController {
     }
 
     public async canAccept(params: DecideRequestParametersJSON): Promise<ValidationResult> {
-        return await this.canDecide({ ...params, accept: true });
+        const canDecideResult = await this.canDecide({ ...params, accept: true });
+
+        const request = await this.getOrThrow(params.requestId);
+
+        const keyUniquenessValidationResult = this.validateKeyUniquenessOfRelationshipAttributesWithinRequest(request.content.items, params.items);
+        if (keyUniquenessValidationResult.isError() && canDecideResult.isSuccess()) {
+            return keyUniquenessValidationResult;
+        }
+
+        return canDecideResult;
```

Comment:
Early return is possible if `canDecideResult` is an error result and I have changed it that way. Returning `keyUniquenessValidationResult` instead of `canDecideResult` in case of both results are success results is not possible as they have different formats. The tests would fail otherwise.